src/main.go

```diff
@@ -0,0 +1,320 @@
+
+	if err != nil {
+		return nil, fmt.Errorf("did not get workload with id %s: %v", *workloadID, err)
+	}
+
+	workloadJSON, err := json.Marshal(workloadOutput.Workload)
+
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling workload with id %s: %v", *workloadID, err)
+	}
+
+	workloadJSON, err = addTypeToData(workloadJSON)
+
+	if err != nil {
+		return nil, err
+	}
+
+	wa.data = append(wa.data, workloadJSON)
+
+	return workloadOutput.Workload, nil
+}
+
+func (wa *WellArchitected) getLensReview(workloadID *string, lensAlias *string) (*wellarchitected.GetLensReviewOutput, error) {
+	lensReviewInput := &wellarchitected.GetLensReviewInput{
+		WorkloadId: workloadID,
+		LensAlias:  lensAlias,
+	}
+
+	lensReviewOutput, err := wa.client.GetLensReview(wa.ctx, lensReviewInput)
+
+	if err != nil {
+		return nil, fmt.Errorf("did not get lens review with workload id %s and lens alias %s: %v", *workloadID, *lensAlias, err)
+	}
+
+	return lensReviewOutput, nil
+}
+
+func (wa *WellArchitected) parseLensReview(lensReview *wellarchitected.GetLensReviewOutput, pillarReviewSummary *types.PillarReviewSummary) error {
+	lensReviewJSON, err := json.Marshal(lensReview)
+
+	if err != nil {
+		return fmt.Errorf("error marshaling lens review with workload id %s and lens alias %s: %v",
+			*lensReview.WorkloadId, *lensReview.LensReview.LensAlias, err)
+	}
+
+	var lensReviewMap map[string]interface{}
+	err = json.Unmarshal(lensReviewJSON, &lensReviewMap)
+
+	if err != nil {
+		return fmt.Errorf("error unmarshaling lens review with workload id %s and lens alias %s: %v",
+			*lensReview.WorkloadId, *lensReview.LensReview.LensAlias, err)
+	}
+
+	delete(lensReviewMap["LensReview"].(map[string]interface{}), "PillarReviewSummaries")
+	lensReviewMap["LensReview"].(map[string]interface{})["PillarReviewSummary"] = pillarReviewSummary
+
+	parsedLensReviewJSON, err := json.Marshal(lensReviewMap)
+
+	if err != nil {
+		return fmt.Errorf("error marshaling lens review with workload id %s and lens alias %s: %v",
+			*lensReview.WorkloadId, *lensReview.LensReview.LensAlias, err)
+	}
+
+	parsedLensReviewJSON, err = addTypeToData(parsedLensReviewJSON)
+
+	if err != nil {
+		return err
+	}
+
+	wa.data = append(wa.data, parsedLensReviewJSON)
+
+	return nil
+}
+
+func (wa *WellArchitected) getLensReviewImprovements(workloadID *string, lensAlias *string) (*wellarchitected.ListLensReviewImprovementsOutput, error) {
+	lensReviewImprovementsInput := &wellarchitected.ListLensReviewImprovementsInput{
+		WorkloadId: workloadID,
+		LensAlias:  lensAlias,
+	}
+
+	lensReviewImprovementsOutput, err := wa.client.ListLensReviewImprovements(wa.ctx, lensReviewImprovementsInput)
+
+	if err != nil {
+		return nil, fmt.Errorf("did not get lens review improvements with workload id %s and lens alias %s: %v", *workloadID, *lensAlias, err)
+	}
+
+	return lensReviewImprovementsOutput, nil
+}
+
+func (wa *WellArchitected) parseLensReviewImprovements(
```

Comment:
Maybe rename the func so that it will indicate that it not only parses the data, but also adds it to `wa`?